src/engraving/dom/cmd.cpp

```diff
@@ -3404,6 +3411,8 @@ void Score::cmdExplode()
                 if (e && e->type() == ElementType::CHORD) {
                     Chord* c = toChord(e);
                     n = std::max(n, int(c->notes().size()));
+                    n = std::max(n, int(c->graceNotesBefore().size()));
+                    n = std::max(n, int(c->graceNotesAfter().size()));
```

Comment:
Despite what the method names suggest, this checks the number of grace _chords_ before/after. I think that isn't really meaningful, is it? You'll probably want to use `graceNotes` to loop over all grace Chords and query `graceChord->notes().size()` on each of them.